libs/url/builder.ts

```diff
@@ -24,6 +24,15 @@ const SIGNATURE_PARAMETER: string = "ik-s";
 const TIMESTAMP_PARAMETER: string = "ik-t";
 const DEFAULT_TIMESTAMP: string = "9999999999";
 
+//used to check if special char is present in string (you'll need to encode it to utf-8 if it does)
+const hasMoreThanAscii = (str = '') => {
+	return str.split('').some((char) => char.charCodeAt(0) > 127);
+}
+
+const encodeStringIfRequired = (str = '') => {
```

Comment:
@ankur-dwivedi Can you please add a few test cases for this change?

Reply:
Tests are added. Resolving.